web/util/date.ts

```diff
@@ -30,6 +26,22 @@ export const formatTimeDuration = (start: Date, duration: number) => {
   return `PT${hours ? hours + 'H' : ''}${minutes ? minutes + 'M' : ''}`;
 };
 
+const differingMeridiem = (d1: Date, d2: Date) =>
+  d1.getHours() >= 12 !== d2.getHours() >= 12;
```

Comment:
This function doesn't take timezone into account. (This was an issue before the SessionDateTime rewrite too, but I unfortunately didn't catch it then.) How about comparing the outputs of `formatInTimeZone` with `aa` format token to see if they're the same or different, if there's no specialized function for it?

For instance, consider this function call:

```
formatTimeRange(parseISO('2022-04-05T09:00:00Z'), 240, 'America/Los_Angeles')
```

This currently yields " 02:00 AM–06:00" (at least on the server, which seems to use UTC, and on my client given that I'm in the Europe/Oslo timezone). After the fixes for my other comments it would show "2:00 AM – 6:00 AM". Expected: "2:00–6:00 AM".

Conversely, `formatTimeRange(parseISO('2022-04-05T18:00:00Z'), 180, 'America/Los_Angeles')` yields "11:00–14:00 PM" (or after accounting for the formatTime issue, "11:00–2:00 PM"), because in UTC/CEST the start/end times both have `getHours() >= 12`. Expected: "11:00 AM – 2:00 PM".

(I picked the examples carefully so that the bug would be reproducable both on server and on client in CEST, but in general the `getHours` use may cause a "Text content did not match" warning because of the different local timezones on client and server.)

Reply:
Wow. Amazing catch, and thank you very much for the fix suggestion as well. :heart: I don't see anything wrong with doing it this way. 9a514b4

I defaulted to `UTC` as `timeZone`. That doesn't matter, because the assumption is that both the compared dates originate from the same timezone, right?

Reply:
No, I'm afraid it does matter, the point is that if `d1` is AM in UTC and `d2` is PM, that's not generally the case in the timezone we care about, and vice versa. It's a bit awkward to set up test content that demonstrates this, but maybe I should. The cases I mentioned above still apply.

I would think `differingMeridian` needs to take a timezone string as an argument. (Not a problem as such to make this parameter default to `UTC`, but `formatTimeRange` needs to pass its `timezone`.